var/spack/repos/builtin/packages/py-tensorflow-probability/package.py

```diff
@@ -48,9 +58,27 @@ class PyTensorflowProbability(Package):
     depends_on("py-dm-tree", when="@0.12:", type=("build", "run"))
 
     # tensorflow_probability/python/__init__.py
-    depends_on("py-tensorflow@2.10:", when="@0.18:", type=("build", "run"))
-    depends_on("py-tensorflow@2.4:", when="@0.12:", type=("build", "run"))
-    depends_on("py-tensorflow@1.14:", when="@0.8:", type=("build", "run"))
+    # TODO: reactivate the JAX versions once the JAX package is available with newer versions
+    #  also add jaxlib as a dependency
```

Comment:
Where do you see JAX/jaxlib listed as dependencies? Where do the versions come from? I would make a new section with a comment for the file that lists the dep versions so we can easily keep track of when they change.

Reply:
It's as much of a dependency as TF itself: TFP can be run with JAX or with TF and doesn't need one of them to run the other, AFAIU. jax is the pure python package, jaxlib is the compiled backend.

The versions come from the release notes of TFP which say against which version they have been tested; I've been propagating this since a while to the conda repository (https://github.com/conda-forge/tensorflow-probability-feedstock/blob/main/recipe/meta.yaml) and pinning the deps there (as conda doesn't have extras, additional deps are included by default).

Therefore, it seems to me as we either include both, TF _and_ jax, or neither. No strong opinion (and as mentioned, I am rather new to spack and not fully familiar with all the controls the end user has, the general vibes), can be fine both ways (in conda-forge we have both deps, but that's a different philosophy and more towards "it should work for the end user correctly")

Reply:
Spack has extras, we call them "variants". So if you want to make them optional, you can. Again, I'll leave it up to you, I'm a PyTorch guy, not a TF guy. I don't know much about the TF ecosystem, but will try to help where I can to maintain it.